connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -143,6 +163,90 @@ public ConnectorStateInfo connectorStatus(String connName) {
                 status.workerId(), status.trace());
     }
 
+
+    @Override
+    public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
+        ConfigDef connectorConfigDef = ConnectorConfig.configDef();
+        List<ConfigValue> connectorConfigValues = connectorConfigDef.validate(connectorConfig);
+        ConfigInfos result = generateResult(connType, connectorConfigDef.configKeys(), connectorConfigValues, Collections.<String>emptyList());
+
+        if (result.errorCount() != 0) {
+            return result;
+        }
+
+        ConnectorConfig connConfig = new ConnectorConfig(connectorConfig);
+        String connName = connConfig.getString(ConnectorConfig.NAME_CONFIG);
+        Connector connector = getConnector(connType, connName);
+
+        Config config = connector.validate(connectorConfig);
+
+        Map<String, ConfigKey> configKeys = config.configDef().configKeys();
+        List<ConfigValue> configValues = config.configValues();
+
+        Map<String, ConfigKey> resultConfigKeys = new HashMap<>(configKeys);
+        resultConfigKeys.putAll(connectorConfigDef.configKeys());
```

Comment:
`resultConfigKeys` is only used in this `putAll`, can we just remove it? It seems like the `configKeys` shouldn't ever depend on the validation -- aren't they static so you can just get them from `connectorConfigDef` like you do on the previous `generateResult` call?